pytest-embedded-idf/pytest_embedded_idf/serial.py

```diff
@@ -84,30 +83,38 @@ def load_ram(self) -> None:
         if self.app.bin_file:
             bin_file = self.app.bin_file
         else:
-            live_print_call(
+            with contextlib.redirect_stdout(self._q):
+                esptool.main(
+                    [
+                        '--chip',
+                        self.app.target,
+                        'elf2image',
+                        self.app.elf_file,
+                        *self.app.write_flash_args,
+                    ],
+                    esp=self.esp,
+                )
+            bin_file = self.app.elf_file.replace('.elf', '.bin')
+
+        with contextlib.redirect_stdout(self._q):
+            esptool.main(
                 [
-                    'esptool.py',
                     '--chip',
                     self.app.target,
-                    'elf2image',
-                    self.app.elf_file,
-                    *self.app.write_flash_args,
+                    '--no-stub',
+                    'load_ram',
+                    bin_file,
                 ],
-                msg_queue=self._q,
+                esp=self.esp,
             )
-            bin_file = self.app.elf_file.replace('.elf', '.bin')
 
-        live_print_call(
-            [
-                'esptool.py',
-                '--chip',
-                self.app.target,
-                '--no-stub',
-                'load_ram',
-                bin_file,
-            ],
-            msg_queue=self._q,
-        )
+    def _force_flag(self):
+        if self.app.esp_flash_force:
```

Comment:
since the `flash` function belongs to `EspSerial`, the `esp_flash_force` should goes together with `serial`.

Could you move it into the `EspSerial.__init__`?